src/chatgpt-api.ts

```diff
@@ -1,5 +1,6 @@
 import ExpiryMap from 'expiry-map'
 import pTimeout, { TimeoutError } from 'p-timeout'
+import { ProxyAgent, setGlobalDispatcher } from 'undici'
```

Comment:
We can't use `undici` directly because this package supports more than just node.js.

E.g., we use the default `fetch` implementation if one's available globally, and we only fallback to `undici` if we have to for node.js 16/17.